components/brave_ads/core/internal/common/url/url_util_internal_unittest.cc

```diff
@@ -0,0 +1,208 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/internal/common/url/url_util_internal.h"
+
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+// npm run test -- brave_unit_tests --filter=BraveAds*
+
+namespace brave_ads::internal {
+
+TEST(BraveAdsUrlUtilInternalTest, ReplaceBraveSchemeWithChromeScheme) {
+  // Act & Assert
+  EXPECT_EQ(GURL("chrome://xyzzy"),
+            ReplaceBraveSchemeWithChromeScheme(GURL("brave://xyzzy")));
+}
+
+TEST(BraveAdsUrlUtilInternalTest, DoNotReplaceNonBraveSchemeWithChromeScheme) {
+  // Act & Assert
+  EXPECT_EQ(GURL("file://xyzzy"),
+            ReplaceBraveSchemeWithChromeScheme(GURL("file://xyzzy")));
+}
+
+TEST(BraveAdsUrlUtilInternalTest, HasUrlSearchQueryNameAndValue) {
+  // Act & Assert
+  EXPECT_TRUE(HasUrlQueryNameWithNonEmptyValue(
+      GURL("https://xyzzy.com/?search=thud"), "search"));
+}
+
+TEST(BraveAdsUrlUtilInternalTest, DoesNotHaveUrlSearchQueryName) {
+  // Act & Assert
+  EXPECT_FALSE(HasUrlQueryNameWithNonEmptyValue(
+      GURL("https://xyzzy.com/?foo=bar"), "search"));
+}
+
+TEST(BraveAdsUrlUtilInternalTest, HasUrlSearchQueryNameAndEmptyValue) {
+  // Act & Assert
+  EXPECT_FALSE(HasUrlQueryNameWithNonEmptyValue(
+      GURL("https://xyzzy.com/?search="), "search"));
+}
+
+TEST(BraveAdsUrlUtilInternalTest, HasSearchQueryNameWithNoValue) {
+  // Act & Assert
+  EXPECT_FALSE(HasUrlQueryNameWithNonEmptyValue(
+      GURL("https://xyzzy.com/?search"), "search"));
+}
+
+TEST(BraveAdsUrlUtilInternalTest, DoesETLDPlusOneContainAsteriskWildcard) {
+  // Act & Assert
+  EXPECT_FALSE(DoesETLDPlusOneContainWildcard(GURL("*.xyzzy.com")));
```

Comment:
there is no test with DoesETLDPlusOneContainWildcard returning true

Reply:
Agreed, let's add `*.appspot.com` which is expected to return `true`.

Reply:
Great catch. Added. Thanks